[MRG+2] DOC examples added to the rest of sklearn/covariance classes

adrinjalali opened the pull request:
See #3846 

Examples added to GraphicalLasso and GraphicalLassoCV in `sklearn/covariance/graph_lasso_.py`

Commit:
added graphicallasso and graphicallassocv examples

jnothman commented on sklearn/covariance/graph_lasso_.py:
```diff
@@ -553,6 +580,33 @@ class GraphicalLassoCV(GraphicalLasso):
+    >>> emp_cov = np.dot(X.T, X) / n_samples
+    >>> model = GraphicalLassoCV(cv=5)
+    >>> loglik_est = -model.fit(X).score(X)
+    >>> loglik_real = -log_likelihood(emp_cov, linalg.inv(cov))
```
Should we use prec rather than invert again?

Commit:
don't double invert

adrinjalali commented:
@jnothman any idea why only one of the travis jobs is failing? It seems like a numerical error, but a large one.

Commit:
reported log_likelihood ...

Commit:
fir real negative log likelihood

jnothman commented:
What kind of differences are we talking about?

adrinjalali commented:
@jnothman 
the estimated negative log likelihood: 26.1847 vs 26.1927
the real negative log likelihood: 27.8393 vs 27.7033

jnothman commented:
ping me about this a while after release and we will try understand if it's
a bug... i suppose it's probably not

adrinjalali committed:
fix graphlasso rand issue

adrinjalali committed:
add elliptic_envelope example

adrinjalali committed:
make the test numerically more stable [I guess]

jnothman commented on sklearn/covariance/elliptic_envelope.py:
```diff
@@ -68,6 +68,24 @@ class EllipticEnvelope(MinCovDet, OutlierMixin):
+    array([[0.7411..., 0.2535...],
+           [0.2535..., 0.3053...]])
+    >>> cov.location_
+    array([0.0813... , 0.0427...])
```
Should we also illustrate the outlier detection interface?

adrinjalali replied:
done

jnothman commented on sklearn/covariance/graph_lasso_.py:
```diff
@@ -327,6 +327,33 @@ class GraphicalLasso(EmpiricalCovariance):
+    ...                               largest_coef=.7,
+    ...                               random_state=0)
+    >>> cov = linalg.inv(prec)
+    >>> d = np.sqrt(np.diag(cov))
```
I think this normalisation deserves a comment, and ideally would be g performed with some existing scipy function

adrinjalali replied:
I found other mistakes in the example. They'd make the example kinda too long, and our docs already show this example. So changed it to be more like the other covariance examples we have, with a manually set sparse covariance matrix, if that's okay.

adrinjalali committed:
show outlier detection for EllipticEnvelope

adrinjalali committed:
change the graphicallasso examples and make them like the others

adrinjalali committed:
remove np.set_printoptions

jnothman committed:
DOC regenerate authors.rst after Adrin has joined (#12719)

adrinjalali committed:
EXA change normed to density in matplotlib calls in examples (#12718)

Commit:
DOC Use defined notation for precision and recall (#12726)

wdevazelhes committed:
ENH Replacing np.where lookup by an inverted index (#9907)

wdevazelhes committed:
FIX warns when invalid n_components in LinearDiscriminantAnalysis (#11526)

TomDLT committed:
MNT cleaning whats_new

jeremiedbb committed:
MNT Remove unnecessary custom nested fused types by builtin floating (#12737)

jnothman committed:
DOC Revert "DOC Use defined notation for precision and recall (#12726)" (#12743)

rth committed:
MAINT Use unicode in feature_extraction tests (#12709)

* Use unicode in feature_extraction tests

* Add a comment for the alef character

* Set the correct number of words in test_vectorizer_unicode

adrinjalali committed:
DOC Fix math typo in model_evaluation's explained_variance (#12747)

JanSellner committed:
DOC Clarify what the decision function in SVM calculates (#12708)

qinhanmin2014 committed:
Revert "DOC Fix math typo in model_evaluation's explained_variance (#12747)"

This reverts commit bf2f648639e92246c4b92ad039c9b086d6e6edce.

albertcthomas committed:
DOC change "cumulative density" to "cumulative distribution" (#12754)

NicolasHug committed:
DOC Added roadmap.rst (#12761)

eamanu committed:
ENH Use accuracy instead of micro-average in classification_report (#12758)

hassaanseeker committed:
Update linear_model.rst (#12735)

* Update for Logistic-Regression Table Summary for L-BFGS

* Update for linear_model.rst Logistic-Regression L-BFGS summary.

* Update for Logistic-Regression Doc.

* Logistic-Regression Update

* Update linear_model.rst

* Update jnothman comments

* Update jnothman comments

* Update for jnothman comments.

* test

* test2

* test3

* test4

* Update for Comments from amueller and jnothman

* Update for Comments from amueller and jnothman

amourav committed:
MNT Deprecate assert_true and assert_false (#12717)

amueller committed:
DOC Explain solver choices for LogisticRegression (#12768)

adrinjalali committed:
Remove python < 3.5 from CI (#12746)

qinhanmin2014 committed:
MNT flake8 in r2_score examples (#12773)

adrinjalali committed:
MNT hotfix circle-ci job dependency (#12780)

adrinjalali committed:
DOC update min required versions in some remaining docs (#12778)

adrinjalali committed:
DOC remove python2.7 from badge list in README (#12781)

qinhanmin2014 committed:
MNT Don't persist doc in doc-min-dependencies job (#12782)

dafeda committed:
EXA Avoiding TypeError when using --all_categories in plot_document_classification_20newsgroups.py (#12770)

qinhanmin2014 committed:
DOC Remove what's new entry for unreleased feature

katrinleinweber committed:
DOC Hyperlink DOIs to preferred resolver (#12792)

Commit:
API (0.21) Handling parameter labels removal from hamming_loss (#12656)

qinhanmin2014 committed:
DOC (0.21) update Circle CI doc

Commit:
DOC Fix LaTeX text{} block escaping of underscores in doc-modules-model_evaluation.rst (#12788)

Commit:
DOC Added description for classes_ in LogisticRegression/LogisticRegressionCV documentation (#12795)

jeremiedbb committed:
FIX pairwise distances with 'seuclidean' or 'mahalanobis' metrics (#12701)

Fixes #12672

amueller committed:
DOC Update what's new according to 0.20.X

amueller committed:
DOC typo simple present singular for SGDClassifier in linear model docs (#12804)

jnothman committed:
DOC reorder what's new 0.20.2

qinhanmin2014 committed:
TST np.vstack won't support generator in the future (#12816)

qinhanmin2014 committed:
DOC Update changed models for 0.20.2 (#12810)

Commit:
DOC Added example to decomposition.DictionaryLearning (#12799)

jnothman committed:
Reduce precision requirements for float32 PCA (#12825)

Fixes #12823

qinhanmin2014 committed:
Revert "DOC Added example to decomposition.DictionaryLearning (#12799)"

This reverts commit fec534a9934d2fe94a259c4582b73cc32eb76e1a.

Commit:
DOC Shorten and improve CONTRIBUTING.md (#12801)

* Reuse intro from contributing.rst

* Removed content already on rst

Simplified CONTRIBUTING.md by removing content that already
exists in contributing.rst. Added Code of Conduct reference
as seen in a few other CONTRIBUTING.md of popular libraries.

* Update contributing.rst

In shortening `CONTRIBUTING.md`, material deleted from `how to contribute` section are added onto `contributing.rst` if it doens't exist on the latter.

* added full link to .rst file + removed 'Full' in 'Full Docum...'

bertrandhaut committed:
DOC bad link in ParameterGrid documentation (#12828)

drorata committed:
FIX Add more validation for size parameters in train_test_split (#12733)

* Fixed typo in an example

* Removed un-needed lines from example

* Added tests on the validity of parameters

* Fixed error msg for train_size issue

* Added tests to check the validation of test and train sizes

* Switched to parameterized test

* Fixed typo in error msg

* Swithced to pytest.raises

* Swithced to pytest.raises also when checking msg

* Improved the validity tests and their unit testing

qinhanmin2014 committed:
DOC add |Feature| tag in what's new

NicolasHug committed:
FIX Fixed gradient computation in gradient boosting for multiclass classification (#12715)

* Fixed gradient update

* Added whatsnew entry

NicolasHug committed:
[MRG+1] Deprecated logistic_regression_path (#12821)

<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md#pull-request-checklist
-->

#### Reference Issues/PRs
<!--
Example: Fixes #1234. See also #3456.
Please use keywords (e.g., Fixes) to create link to the issues or pull requests
you resolved, so that they will automatically be closed when your pull request
is merged. See https://github.com/blog/1506-closing-issues-via-pull-requests
-->

Closes #12798 

#### What does this implement/fix? Explain your changes.

This PR deprecates the use of `logistic_regression_path` and makes it private.

#### Any other comments?


<!--
Please be aware that we are a loose team of volunteers so patience is
necessary; assistance handling other issues is very welcome. We value
all user contributions, no matter how minor they are. If we are slow to
review, either the pull request needs some benchmarking, tinkering,
convincing, etc. or more likely the reviewers are simply busy. In either
case, we ask for your understanding during the review process.
For more information, see our FAQ on this topic:
http://scikit-learn.org/dev/faq.html#why-is-my-pull-request-not-getting-any-attention.

Thanks for contributing!
-->

thomasjpfan committed:
TST Adds multilabels permutation tests to metrics/test_common (#12803)

NicolasHug committed:
[MRG] Add pprint for estimators - continued (#11705)

* add pprint for estimators

* strip color from length, add color option

* Minor cleaning, fixes, factoring and docs

* Added some basic tests

* Fixed line length issue

* fixed flake8 and added visual test for review

* Fixed test

* Fixed Python 2 issues (inspect.signature import)

* Trying to fix flake8 again

* Added special repr for functions

* Added some other visual tests

* Changed _format_function in to _format_callable

because callable() returns True also for class objects (which we want to
reprensent with their name as well anyway)

* Consistent output in Python 2 and 3

* WIP

* Now using the builtin pprint module

* pep8

* Added changed_only param

* Fixed printing when string would fit in less than line width

* Fixed printing of steps parameter

* Fixed changed_only param for short estimators

* fixed pep8

* Added some more description in docstring

* changed_only is now an option from set_config()

* Put _pprint.py into sklearn/utils, added tests

* Added doctest NORMALIZE_WHITESPACE where needed

* Fixed tests

* fix test-doc

* fixing test that passed before....

* Fixed tests

* Added test for changed_only and long lines

* typo

* Added authors names

* Added license file

* Added ellipsis based on number of elements in sequence + added increasinly aggressive repr strategies

* Updated whatsnew

* dont use increaingly aggressive strategy

* Fixed tests

* Removed LICENSE file and put license text in _pprint.py

* fixed test_base

* Sorted parameters dictionary for consistent output in 3.5

* Actually using OrderedDict...

* Addressed comments

* Added test for NaN changed parameter

* Update whatsnew

* Added example to set_config()

* Removed example

* Added example in gallery

* Spelling

adrinjalali committed:
CI move to the latest cython in ci (#12829)

jnothman committed:
BLD add a mac run on Travis (#12824)

thomasjpfan committed:
MAINT Remove minor duplication in metrics.__init__ (#12851)

akahard2dj committed:
DOC improving import convenience in class examples (#12846)

eamanu committed:
DOC Document that fetch_20newsgroups also returns target_names (#12783)

rth committed:
BENCH Make benchmarks/bench_text_vectorizers.py run faster (#12842)

akahard2dj committed:
DOC Resolve a missing link on the home page (#12847)

NicolasHug committed:
DOC Fixed typo in contributors guide (#12868)

amueller committed:
DOC better explain bootstrap option in forests and bagging (#12875)

adrinjalali committed:
BLD circle-ci should only run/build plot_* files (#12797)

albertcthomas committed:
DOC add sentence about line length of rst files (#12883)

amueller committed:
DOC link to dev docs instead of contributing docs in the menu (#12885)

NicolasHug committed:
DOC Added tips for reading the code base (#12874)

* Added tips for reading the code base

* Put it in contributing.rst

* Added bullet point about inheritance

qinhanmin2014 committed:
TST Manually scramble the indices in svm tests (#12890)

* [scipy-dev] manually scramble the indices

* [scipy-dev] review comment

elsander committed:
FIX predict method for multiclass multioutput ensemble models (#12834)

pakallis committed:
DOC Fix typo in glossary.rst (#12913)

amueller committed:
MRG Drop legacy python / remove six dependencies (#12639)

rth committed:
MNT Use scipy.special.xlogy to avoid indefinite limit in 0 for x*log(y)  (#12915)

ZaydH committed:
FIX Use scipy.special.expit in calibration (#12909)

qinhanmin2014 committed:
FIX learning_rate default value in update_terminal_regions (#12925)

BoboTiG committed:
EXA Fix several DeprecationWarning: invalid escape sequence in examples (#12924)

GaelVaroquaux committed:
DOC How to deal with stalled PR (#12894)

rth committed:
MAINT Use set litterals when possible (#12667)

adrinjalali committed:
(review reminder) Merge remote-tracking branch 'upstream/master' into examples/graphlasso

ogrisel commented on sklearn/covariance/elliptic_envelope.py:
```diff
@@ -68,6 +68,28 @@ class EllipticEnvelope(MinCovDet, OutlierMixin):
+    --------
+    >>> import numpy as np
+    >>> from sklearn.covariance import EllipticEnvelope
+    >>> from sklearn.datasets import make_gaussian_quantiles
```
Not used.

ogrisel commented on sklearn/covariance/elliptic_envelope.py:
```diff
@@ -68,6 +68,28 @@ class EllipticEnvelope(MinCovDet, OutlierMixin):
+    >>> import numpy as np
+    >>> from sklearn.covariance import EllipticEnvelope
+    >>> from sklearn.datasets import make_gaussian_quantiles
+    >>> real_cov = np.array([[.8, .3],
```
nitpick: maybe `true_cov`?

adrinjalali committed:
Merge remote-tracking branch 'upstream/master' into examples/graphlasso

ogrisel commented on sklearn/covariance/elliptic_envelope.py:
```diff
@@ -68,6 +68,28 @@ class EllipticEnvelope(MinCovDet, OutlierMixin):
+    >>> np.random.seed(0)
+    >>> X = np.random.multivariate_normal(mean=[0, 0],
+    ...                                   cov=real_cov,
+    ...                                   size=500)
```
Personally I prefer the object oriented idiom for seeding:

```python
    >>> rng = np.random.RandomState(0)
    >>> X = rng.multivariate_normal(...
```

or just:

```python
    >>> X = np.random.RandomState(0).multivariate_normal(
    ...     mean=[0, 0], cov=real_cov, size=500)
```

adrinjalali committed:
address comments

jnothman reviewed (approved):
Pending real_ -> true_

adrinjalali committed:
real_conv -> true_conv

ogrisel reviewed (approved):
Let's merge.
